src/praisonai-agents/praisonaiagents/context/compressor.py

```
@@ -0,0 +1,467 @@
+                fallback_summary = self._create_fallback_summary(middle)
+                summary_msg = {
+                    "role": "system",
+                    "content": f"[Fallback Summary]\n{fallback_summary}",
+                    "_compression_metadata": {
+                        "original_message_count": len(middle),
+                        "fallback": True,
+                        "compressed_at": datetime.utcnow().isoformat(),
+                        "session_id": session_id,
+                    }
+                }
+                compressed_messages = head + [summary_msg] + tail
```

Ruff (0.15.15)

[warning] Consider `[*head, summary_msg, *tail]` instead of concatenation

Replace with `[*head, summary_msg, *tail]`

(RUF005)